src/List.mo

```diff
@@ -499,43 +499,35 @@ module {
     }
   };
 
-  /// Returns the element at index `index`. Indexing is zero-based.
-  /// Traps if `index >= size`, error message may not be descriptive.
+  /// Returns the element at index `index` as an option.
+  /// Returns `null` when `index >= size`. Indexing is zero-based.
   ///
   /// Example:
   /// ```motoko include=import
   /// let list = List.empty<Nat>();
   /// List.add(list, 10);
   /// List.add(list, 11);
-  /// assert List.get(list, 0) == 10;
+  /// assert List.get(list, 0) == ?10;
+  /// assert List.get(list, 2) == null;
   /// ```
   ///
   /// Runtime: `O(1)`
-  public func get<T>(list : List<T>, index : Nat) : T {
-    // inlined version of:
-    //   let (a,b) = locate(index);
-    //   switch(list.blocks[a][b]) {
-    //     case (?element) element;
-    //     case (null) Prim.trap "";
-    //   };
-    let i = Nat32.fromNat(index);
-    let lz = Nat32.bitcountLeadingZero(i);
-    let lz2 = lz >> 1;
-    switch (
-      if (lz & 1 == 0) {
-        list.blocks[Nat32.toNat(((i << lz2) >> 16) ^ (0x10000 >> lz2))][Nat32.toNat(i & (0xFFFF >> lz2))]
-      } else {
-        list.blocks[Nat32.toNat(((i << lz2) >> 15) ^ (0x18000 >> lz2))][Nat32.toNat(i & (0x7FFF >> lz2))]
-      }
-    ) {
-      case (?result) return result;
-      case (_) Prim.trap "List index out of bounds in get"
+  ///
+  /// Space: `O(1)`
+  public func get<T>(list : List<T>, index : Nat) : ?T {
+    let (a, b) = locate(index);
+    if (a < list.blockIndex or list.elementIndex != 0 and a == list.blockIndex) {
+      list.blocks[a][b]
+    } else {
+      null
     }
   };
 
   /// Returns the element at index `index` as an option.
   /// Returns `null` when `index >= size`. Indexing is zero-based.
   ///
+  /// **Deprecated**: This function is now identical to `get()`. Use `get()` instead.
+  ///
```

Comment:
Can we just rename `get -> at` and `getOpt -> get`? Without any other changes. I'd leave the inlined implementation in `at`, why change that now?
I think that would be a cleaner approach

Reply:
I agree. So `at` for trapping, and `get` for option returning